src/check.ml

```diff
@@ -641,8 +641,8 @@ and checkInit  (i: init) : typ =
                 match elen with
                 | None -> 0L
                 | Some e -> (ignore (checkExp true e);
-                match isInteger (constFold true e) with
-                  Some len -> len
+                match getInteger (constFold true e) with
+                  Some len -> Z.to_int64 len (* Z on purpose, we don;t want to ingore overflows here *)
```

Comment:
Why are we still converting to `int64` here (and below)? What happens if it overflows?

Reply:
If it overflows, it throws instead of just getting the lower bits.
It is comparing it with the length of the initializer here, and that in itself is an `int64` that would silently overflow.

I think there are a lot of other problems with arrays that have a length longer than the largest positive number that can fit into an `int64`.